pageserver/src/tenant.rs

```diff
@@ -493,6 +493,8 @@ pub struct OffloadedTimeline {
     pub tenant_shard_id: TenantShardId,
     pub timeline_id: TimelineId,
     pub ancestor_timeline_id: Option<TimelineId>,
+    /// Whether to retain the branch lsn at the ancestor or not
+    pub ancestor_retain_lsn: Option<Lsn>,
```

Comment:
nit: These optionals will either both be set or both be None. Perhaps it would make more sense to only have one that holds a tuple?

Reply:
For now, your statement is true. In the future, there will be cases where `ancestor_timeline_id` can be `Some(_)` but `ancestor_retain_lsn` is `None`, because the timeline has already been flattened.

If you want, I can put the two into a tuple and refer to a future refactor to put optionals. Please write what you would prefer.